kernel/libs/aster-bigtcp/src/iface/poll_iface.rs

```diff
@@ -0,0 +1,287 @@
+// these `pub(crate)` methods in the future (FIXME).
+impl<'a, E: Ext> PollableIfaceMut<'a, E> {
+    pub(crate) fn new(
+        context: &'a mut smoltcp::iface::Context,
+        pending_conns: &'a mut PendingConnSet<E>,
+    ) -> Self {
+        Self {
+            context,
+            pending_conns,
+        }
+    }
+
+    pub(crate) fn inner_mut(&mut self) -> (&mut smoltcp::iface::Context, &mut PendingConnSet<E>) {
+        (self.context, self.pending_conns)
+    }
+}
+
+impl<E: Ext> PollableIfaceMut<'_, E> {
+    pub(super) fn pop_pending_tcp(&mut self) -> Option<Arc<TcpConnectionBg<E>>> {
+        let now = self.context.now.total_millis() as u64;
+        self.pending_conns.pop_tcp_before_ms(now)
+    }
+}
+
+impl<E: Ext> PollableIfaceMut<'_, E> {
+    /// Returns an immutable reference to the `smoltcp` context.
+    pub(crate) fn context(&self) -> &smoltcp::iface::Context {
+        self.context
+    }
+
+    /// Returns the `smoltcp` context for passing to the `smoltcp` APIs.
+    pub(crate) fn context_mut(&mut self) -> &mut smoltcp::iface::Context {
+        self.context
+    }
+
+    /// Updates the next poll time of `socket` to `poll_at`.
+    ///
+    /// This method (or [`PollableIface::update_next_poll_at_ms`]) should be called after network
+    /// or user events that change the poll time occur.
+    pub(crate) fn update_next_poll_at_ms(
+        &mut self,
+        socket: &Arc<TcpConnectionBg<E>>,
+        poll_at: smoltcp::socket::PollAt,
+    ) -> NeedIfacePoll {
+        self.pending_conns.update_next_poll_at_ms(socket, poll_at)
+    }
+}
+
+/// A key to sort sockets by their next poll time.
+pub(crate) struct PollKey {
+    next_poll_at_ms: AtomicU64,
+    id: usize,
+}
+
+impl PartialEq for PollKey {
+    fn eq(&self, other: &Self) -> bool {
+        self.next_poll_at_ms.load(Ordering::Relaxed)
+            == other.next_poll_at_ms.load(Ordering::Relaxed)
+            && self.id == other.id
+    }
+}
+impl Eq for PollKey {}
+impl PartialOrd for PollKey {
+    fn partial_cmp(&self, other: &Self) -> Option<core::cmp::Ordering> {
+        Some(self.cmp(other))
+    }
+}
+impl Ord for PollKey {
+    fn cmp(&self, other: &Self) -> core::cmp::Ordering {
+        self.next_poll_at_ms
+            .load(Ordering::Relaxed)
+            .cmp(&other.next_poll_at_ms.load(Ordering::Relaxed))
+            .then_with(|| self.id.cmp(&other.id))
+    }
+}
+
+impl PollKey {
+    /// Creates a new [`PollKey`].
+    ///
+    /// `id` must be a unique identifier for the associated socket, as it will be used to locate
+    /// the socket to update its next poll time. This is usually done using the address of the
+    /// [`Arc`] socket (see [`Arc::as_ptr`]).
+    ///
+    /// [`Arc`]: alloc::sync::Arc
+    /// [`Arc::as_ptr`]: alloc::sync::Arc::as_ptr
+    pub(crate) fn new(id: usize) -> Self {
+        Self {
+            next_poll_at_ms: AtomicU64::new(u64::MAX),
```

Comment:
Defining `u64::MAX` as a constant within `PollKey` might be a better approach. Any instance where it is used should then reference this constant instead.

```rust
impl PollKey {
    pub const INACTIVE_VAL: u64 = u64::MAX;
}
```